doc/source/train/user-guides/persistent-storage.rst

```diff
@@ -381,22 +381,29 @@ Note that this behavior is off by default.
 Advanced configuration
 ----------------------
 
-Setting the intermediate local directory
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Setting the local staging directory
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-When a ``storage_path`` is specified, training outputs are saved to an
-*intermediate local directory*, then persisted (copied/uploaded) to the ``storage_path``.
-By default, this intermediate local directory is a sub-directory of ``~/ray_results``.
+Apart from files written to the ``storage_path``, Ray Train also writes to some
+logfiles and metadata files in a *local staging directory* before they get
+persisted (copied/uploaded) to the ``storage_path``.
+By default, this local staging directory is a sub-directory of the temporary Ray session
+directory (``/tmp/ray/session_latest``), which is also where Ray Core logs are dumped.
 
-Customize this intermediate local directory with the ``RAY_AIR_LOCAL_CACHE_DIR`` environment variable:
+Customize the location of the staging directory by customizing the location of the
+temporary Ray session directory. See :ref:`temp-dir-log-files` for how to set this directory.
 
-.. testcode::
-    :skipif: True
+.. warning::
 
-    import os
-    os.environ["RAY_AIR_LOCAL_CACHE_DIR"] = "/tmp/custom/"
+    Prior to 2.10, the ``RAY_AIR_LOCAL_CACHE_DIR`` environment variable was the way to configure
+    the local staging directory to be outside of the home directory (``~/ray_results``).
+
+    ``RunConfig(local_dir)`` was another deprecated alternative to set the local staging directory.
+
+    These options are now deprecated, and you should instead set ``storage_path``.
+    When ``storage_path`` is set, there will no longer be any files written to
+    ``~/ray_results`` starting from Ray 2.10.
```

Comment:
This is a bit confusing since there's a bunch of branching logic based on env var / storage path. Could we maybe simplify this to just focus on the staging directory being set by `RAY_AIR_LOCAL_CACHE_DIR` or `local_dir`? And not talk about `storage_path`?

Reply:
I kept the mention about switching to storage path as a migration path, because users were setting ONLY the environment variable. But now that the env variable doesn't do anything, they need to switch to setting storage path.